pycbc/inference/sampler/dynesty.py

```diff
@@ -0,0 +1,270 @@
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for using the dynesty sampler
+packages for parameter estimation.
+"""
+
+
+from __future__ import absolute_import
+
+import logging
+from pycbc.pool import choose_pool
+import numpy
+import dynesty
+from dynesty.utils import resample_equal
+from pycbc.inference.io import (DynestyFile, validate_checkpoint_files)
+from pycbc.distributions import read_constraints_from_config
+from .base import BaseSampler
+from .base_mcmc import get_optional_arg_from_config
+from .. import models
+
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+class DynestySampler(BaseSampler):
+    """This class is used to construct an Dynesty sampler from the dynesty
+    package.
+
+    Parameters
+    ----------
+    model : model
+        A model from ``pycbc.inference.models``.
+    nlive : int
+        Number of live points to use in sampler.
+    dlogz: float
+        Tolerance limit to the value of logz (also a convergence criteria)
+    pool : function with map, Optional
+        A provider of a map function that allows a function call to be run
+        over multiple sets of arguments and possibly maps them to
+        cores/nodes/etc.
+    """
+    name = "dynesty"
+    _io = DynestyFile
+
+    def __init__(self, model, nlive, dlogz, nprocesses=1,
+                 loglikelihood_function=None, use_mpi=False, **kwargs):
+        self.model = model
+        # Set up the pool
+        model_call = DynestyModel(model, loglikelihood_function)
+        if nprocesses > 1:
+            # these are used to help paralleize over multiple cores / MPI
+            models._global_instance = model_call
+            log_likelihood_call = _call_global_loglikelihood
+            prior_call = _call_global_logprior
+        else:
+            prior_call = model_call.prior_transform
+            log_likelihood_call = model_call.log_likelihood
+        pool = choose_pool(mpi=use_mpi, processes=nprocesses)
+        if pool is not None:
+            pool.size = nprocesses
+
+        self.nlive = nlive
+        self.dlogz = dlogz
+        self.names = model.sampling_params
+        self.ndim = len(model.sampling_params)
+        self.checkpoint_file = None
+        self._sampler = dynesty.NestedSampler(log_likelihood_call,
+                                              prior_call, self.ndim,
+                                              nlive=self.nlive,
+                                              dlogz=self.dlogz,
+                                              pool=pool, **kwargs)
+
+    def run(self):
+        res = self._sampler.run_nested()
```

Comment:
As noted by code climate, the `res` isn't used; just do `self._sampler.run_nested()` instead.